### ISSUE TITLE
llx_paiement_facture: missing lines after upgrading from 17.0.2 to 19.0.2

### ISSUE THREAD

**Issue by syrinx75** [Dolibarr/dolibarr#30620]
### Bug

After the migration from the version 17.0.2 to 19.0.2, we lost many lines in the table llx_paiement_facture.

For these payments, the link between the invoices and the payments have disappeared.

It seems to be a recurring problem for several years.

### Dolibarr Version

19.0.2

### Environment PHP

8.2.21

### Environment Database

MySQL or MariaDB 8.0.36-28

### Steps to reproduce the behavior and expected behavior

Has appended with OVH which seems to work slowly.

### Attached files

<img width="831" alt="image" src="https://github.com/user-attachments/assets/b252097f-2682-4f4b-8720-8335d17e1221">
<img width="954" alt="image" src="https://github.com/user-attachments/assets/f17f6fab-4aaf-4ab3-be18-22e2183c3298">

**Comment by ems-co** [comment]
Did manage to take a backup before upgrading? Try to go back to version 17 and upgrade to 18 and to 19, version by version and not directly to 19

**Comment by syrinx75** [comment]
Yes, we have a backup file. But we work with this broken database for 3 weeks. Now, it seems difficult to restart from an old database.

**Comment by ems-co** [comment]
That's really a serious issue

**Comment by hregis** [comment]
@syrinx75 there is nothing in the update from 17 to 19 that could delete rows in llx_paiement_facture ! Did you update any external modules that could interact with this table ? Anyway you could just restore the missing rows in this table !

**Comment by hregis** [comment]
@syrinx75 you can create a table llx_paiement_facture_old and restore the old table. And use this sql request (foreign key checking may need to be disabled) : 

`INSERT INTO llx_paiement_facture SELECT * FROM llx_paiement_facture_old as pfo WHERE pfo.rowid NOT IN (SELECT pf2.rowid FROM llx_paiement_facture as pf2)`

**Comment by syrinx75** [comment]
> @syrinx75 you can create a table llx_paiement_facture_old and restore the old table. And use this sql request (foreign key checking may need to be disabled) :
> 
> `INSERT INTO llx_paiement_facture SELECT * FROM llx_paiement_facture_old as pfo WHERE pfo.rowid NOT IN (SELECT pf2.rowid FROM llx_paiement_facture as pf2)`

Thanks. I fixed it with this formula. No need to disable the foreign key checking .
